src/main/java/org/cactoos/http/HtResponseEnvelope.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.cactoos.Input;
+
+/**
+ * The envelope for HTTP response.
+ *
+ * @since 0.1
+ */
+public class HtResponseEnvelope implements Input {
+
+    /**
+     * The wire.
+     */
+    private final Wire wire;
+
+    /**
+     * HTTP request.
+     */
+    private final Input request;
+
+    /**
+     * Ctor.
+     * @param wre The wire
+     * @param req The request
+     */
+    public HtResponseEnvelope(final Wire wre, final Input req) {
```

Comment:
@dgroup this should take an Input and delegate to it, the wire and request should stay in HtResponse!

Reply:
@victornoel its unclear for me. 
Please explain why and for what purpose.

Reply:
@dgroup see other examples of Envelopes, you can also check out https://github.com/victornoel/eo-envelopes that contains example as well as a link to yegor's blog on the subject. The envelope shouldn't have behaviour, just wrapping and delegatig an object of its own type.